src/main/java/com/pinterest/secor/parser/DateMessageParser.java

```diff
@@ -39,20 +39,26 @@
  * 
  */
 public class DateMessageParser extends MessageParser {
+
     private static final Logger LOG = LoggerFactory.getLogger(DateMessageParser.class);
     protected static final String defaultDate = "dt=1970-01-01";
     protected static final String defaultFormatter = "yyyy-MM-dd";
     protected SimpleDateFormat outputFormatter = new SimpleDateFormat(defaultFormatter);
     protected Object inputPattern;
     protected SimpleDateFormat inputFormatter;
-    
+
+    protected final String mDtPrefix;
+
     public DateMessageParser(SecorConfig config) {
         super(config);
+
         TimeZone timeZone = config.getTimeZone();
         inputPattern = mConfig.getMessageTimestampInputPattern();
         inputFormatter = new SimpleDateFormat(inputPattern.toString());
         inputFormatter.setTimeZone(timeZone);
         outputFormatter.setTimeZone(timeZone);
+
+        mDtPrefix = (TimestampedMessageParser.usingDatePrefix(config) == null) ? "" : TimestampedMessageParser.usingDatePrefix(config);
```

Comment:
TimestampedMessageParser.usingDatePrefix() already provides a default value, I think you can simply do:

    mDtPrefix = TimestampedMessageParser.usingDatePrefix(config)